src/App.js

```diff
@@ -1,18 +1,37 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react'
+import TodoList from './TodoList'
+import AddTodoForm from './AddTodoForm'
 
+function useSemiPersistentState(key, initialState) {
+	const savedValue = localStorage.getItem(key) || initialState
+	console.log(savedValue)
+
+	const [value, setValue] = useState(JSON.parse(savedValue))
+
+	useEffect(() => {
+		localStorage.setItem(key, JSON.stringify(value))
+	}, [key, value])
+	return [value, setValue]
+}
 function App() {
-  return (
-    <div style={{ textAlign: 'center' }}>
-      <header>
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+	const [todoList, setTodoList] = useSemiPersistentState('savedTodoList', [])
+
+	function removeTodo(id) {
+		const updatedTodoListArr = todoList.filter((item) => item.id !== id)
```

Comment:
Nice job building this function! Consider removing "Arr" from your variable name. It's not necessary to specify what kind of data structure is contained within a variable. Given good variable naming this kind of information should be inferred from the name.
By naming something "list" it's implied that it's a collection of things that naturally belong in an array. The same goes for using a "plural" naming convention. e.g. `todos` implies a list of todos as opposed to `todo` which would be a singular object.